lightning/src/ln/channel.rs

```diff
@@ -1231,7 +1244,7 @@ impl<Signer: Sign> Channel<Signer> {
 		htlcs_included.sort_unstable_by_key(|h| h.0.transaction_output_index.unwrap());
 		htlcs_included.append(&mut included_dust_htlcs);
 
-		(tx, feerate_per_kw, num_nondust_htlcs, htlcs_included)
+		(tx, feerate_per_kw, num_nondust_htlcs, htlcs_included, value_to_self_msat as u64, value_to_remote_msat as u64)
```

Comment:
We should probably document the new return values, emphasizing that its the anticipated value *not* the value actually paid, which may be zero if it got rounded to dust. That said, maybe we actually want to return zero if something was below the dust limit? Should be pretty easy to do - change `value_to_a` and `value_to_b` to `&mut u64` and make them references to the values instead of copies.

Reply:
Added document on `build_commitment_transaction`.

So I thought for a while, and I think that returning post-dust threshold values (which are also post-fee) might falsify our further checks in `send_htlc`. Especially the overdrawing one ("Cannot send value that would overdraw remaining funds"), a sent HTLC might be a dust, thus not inflating further the fee paid, while still being inferior to the remaining balance. Returning a trimmed-to-0 value would lead us to fail this check.

I think it doesn't matter, as ultimately the `dust_limit_satoshis` must be under the channel reserve per-spec compliance, so we fail the chan reserve check if the balance is inferior to `dust_limit_satoshis`. Even further I think we should just combine the 3 last checks in `send_htlc`, if we can satisfy the 2 first ones by transitivity we can't satisfy the current latest one.

Reply:
> Especially the overdrawing one ("Cannot send value that would overdraw remaining funds"), a sent HTLC might be a dust, thus not inflating further the fee paid, while still being inferior to the remaining balance. Returning a trimmed-to-0 value would lead us to fail this check.

Hmm, I feel like that would be correct behavior? If we are currently trimmed to dust we *definitely* shouldn't be sending an additional HTLC?

Reply:
Addressed with `CommitmentStats` introduction, balances are considering dust limit.